components/export.tsx

```diff
@@ -1,33 +1,32 @@
+import React from 'react';
 import Link from 'next/link';
-
 import TemplateComponentProps from '@/templates/props';
-import QuestionPaper from '@/templates/question-paper';
-
-export const ExportAsPDF = async ({
-  children,
-  className,
-  data,
-}: {
-  children: React.ReactNode;
-  className?: string;
-  data: TemplateComponentProps;
-}) => {
-  const ReactDOMServer = (await import('react-dom/server')).default;
-  const htmlBody = ReactDOMServer.renderToStaticMarkup(
-    <QuestionPaper data={data} />
-  );
+import PDFTemplate from './PDFTemplate';
 
-  // Generate PDF with the above HTML
+const ExportAsPDF = ({ data }: { data: TemplateComponentProps }) => {
+  // Generate PDF with the question paper data
+  const generatePDF = () => {
+    return (
+      <PDFTemplate data={data} />
+    );
+  };
 
-  // Store the PDF in a temporary cache on our S3 bucket
+  // Dummy function for storing the PDF in a S3 bucket 
+  const storePDFInS3 = async () => {
+    
+    return (
+      <Link href="#" target="_blank">
+        View PDF from S3
+      </Link>
+    );
+  };
 
   return (
-    <Link
-      className={className}
-      href={`https://blob.nitkkr.ac.in/cache/path/to/file.pdf`}
-      target="_blank"
-    >
-      {children}
-    </Link>
+    <div>
+      {generatePDF()}
+      {storePDFInS3()} 
+    </div>
```

Comment:
Using, `<PDFTemplate />` means it will return the PDF as a *component*, how will we pass this PDF to our function that is supposed to save this PDF? This is only good if we want to render the PDF on a page.

Reply:
I can try generating pdf bytes and creating a blob object instead.

Reply:
The pdf will open in a new window instead of being rendered as a component.